src/App.tsx

```diff
@@ -1,60 +1,153 @@
+    }
+  };
+
+  const handleSelectUser = useCallback((user: User) => {
+    if (user.id !== selectedUserId) {
+      setSelectedUserId(user.id);
+      setSelectedPostId(0);
+      setPosts([]);
+    }
+  }, [selectedUserId]);
+
+  const selectedPost = useMemo(() => {
+    return posts.find(({ id }) => id === selectedPostId);
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    if (selectedUserId) {
+      fetchUserPosts();
+    }
+  }, [selectedUserId]);
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
 
-export const App: React.FC = () => {
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  onUserSelect={handleSelectUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUserId && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {noPosts && (
+                  <div
+                    className="notification is-warning"
+                    data-cy="NoPostsYet"
+                  >
+                    No posts yet
+                  </div>
+                )}
 
-                <PostsList />
+                {posts.length > 0 && (
+                  <PostsList
+                    posts={posts}
+                    selectedPostId={selectedPostId}
+                    onPostSelect={setSelectedPostId}
+                  />
+                )}
               </div>
             </div>
           </div>
 
           <div
             data-cy="Sidebar"
-            className={classNames(
+            className={cn(
               'tile',
               'is-parent',
               'is-8-desktop',
               'Sidebar',
-              'Sidebar--open',
+              { 'Sidebar--open': selectedPostId > 0 },
```

Comment:
In the real world, post id can be zero, so check my previous comment and here it can be updated too.